From 04a74154e1e2cb0faa8dd80b62274c78c589ce03 Mon Sep 17 00:00:00 2001
From: Billy Laws <blaws05@gmail.com>
Date: Wed, 17 Apr 2024 18:50:05 +0000
Subject: [PATCH 03/14] ntdll: Force all wow64 host allocations outside of the
 32-bit AS.

---
 dlls/ntdll/unix/virtual.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index 0c0a12c394d..b67c454efdf 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -4569,6 +4569,9 @@ static NTSTATUS allocate_virtual_memory( void **ret, SIZE_T *size_ptr, ULONG typ
     {
         base = NULL;
         size = (size + page_mask) & ~page_mask;
+#ifdef _WIN64
+        if (is_wow64() && (!limit_high || limit_high > limit_4g)) limit_low = limit_4g;
+#endif
     }
 
     /* Compute the alloc type flags */
-- 
2.48.1

