From b228663d1dec05ca268e5758746e1244e4dedfb3 Mon Sep 17 00:00:00 2001
From: Billy Laws <blaws05@gmail.com>
Date: Fri, 27 Sep 2024 11:29:21 +0000
Subject: [PATCH 09/14] HACK: define some extra arm64ec symbols to workaround
 llvm bugs

---
 aclocal.m4 |   6 ++
 configure  | 226 ++++++++++++++++++++++++++++++++++++++++++++++++++++-
 2 files changed, 230 insertions(+), 2 deletions(-)

diff --git a/aclocal.m4 b/aclocal.m4
index 9efbd7ebb3e..dacc90a4255 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -201,6 +201,12 @@ AC_LINK_IFELSE([AC_LANG_SOURCE([[void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }]])],
                [AS_VAR_SET(ac_var,yes)], [AS_VAR_SET(ac_var,no)])
 CFLAGS=$ac_wine_try_cflags_saved
diff --git a/configure b/configure
index beba1346b6a..558921cac19 100755
--- a/configure
+++ b/configure
@@ -10413,6 +10413,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -10453,6 +10459,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -10672,6 +10684,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -10711,6 +10729,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -10750,6 +10774,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -10789,6 +10819,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -10828,6 +10864,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -10867,6 +10909,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -10906,6 +10954,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -10945,6 +10999,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -10984,6 +11044,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11023,6 +11089,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11062,6 +11134,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11101,6 +11179,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11140,6 +11224,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11179,6 +11269,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11218,6 +11314,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11257,6 +11359,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11296,6 +11404,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11335,6 +11449,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11374,6 +11494,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11413,6 +11539,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11452,6 +11584,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11493,6 +11631,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11533,6 +11677,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11574,6 +11724,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11615,6 +11771,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11654,6 +11816,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11693,6 +11861,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11732,6 +11906,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11771,6 +11951,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11810,6 +11996,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11849,6 +12041,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11911,6 +12109,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11951,6 +12155,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -11990,6 +12200,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -12033,6 +12249,12 @@ void *__os_arm64x_dispatch_ret = 0;
 #if defined(__clang_major__) && defined(MIN_CLANG_VERSION) && __clang_major__ < MIN_CLANG_VERSION
 #error Too old clang version
 #endif
+void *__os_arm64x_dispatch_call_no_redirect = 0;
+void *memcpy() {};
+void *__os_arm64x_check_icall =0;
+void *memset() {};
+void *memmove() {};
+
 int __cdecl mainCRTStartup(void) { return 0; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
@@ -21948,7 +22170,7 @@ esac
 
 
 enable_win16=${enable_win16:-i386}
-enable_vcruntime140_1=${enable_vcruntime140_1:-x86_64,arm64ec}
+enable_vcruntime140_1=${enable_vcruntime140_1:-x86_64,arm64ec,aarch64}
 
 if test -n "$PE_ARCHS"
 then
@@ -21965,7 +22187,7 @@ fi
 
 enable_cmd=${enable_cmd:-yes}
 enable_dllhost=${enable_dllhost:-yes}
-enable_dpnsvr=${enable_dpnsvr:-i386,x86_64,arm64ec}
+enable_dpnsvr=${enable_dpnsvr:-i386,x86_64,arm64ec,aarch64}
 enable_dxdiag=${enable_dxdiag:-yes}
 enable_msiexec=${enable_msiexec:-yes}
 enable_netsh=${enable_netsh:-yes}
-- 
2.48.1

