From f6587a10e87a001e95b6fcfa9152e55bb93b808d Mon Sep 17 00:00:00 2001
From: Billy Laws <blaws05@gmail.com>
Date: Fri, 12 Apr 2024 20:54:16 +0000
Subject: [PATCH 02/14] HACK: server: Map within the 39-bit address space on
 arm64.

---
 server/mapping.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/server/mapping.c b/server/mapping.c
index 4bdd7c5080a..2ddfbac11e6 100644
--- a/server/mapping.c
+++ b/server/mapping.c
@@ -261,7 +261,11 @@ void init_memory(void)
 {
     page_mask = sysconf( _SC_PAGESIZE ) - 1;
     free_map_addr( 0x60000000, 0x1c000000 );
+#ifdef __aarch64__
+    free_map_addr( 0x4000000000, 0x1000000000 );
+#else
     free_map_addr( 0x600000000000, 0x100000000000 );
+#endif
     if (page_mask != 0xfff)
         fprintf( stderr, "wineserver: page size is %uk but Wine requires 4k pages, expect problems\n",
                  (int)(page_mask + 1) / 1024 );
-- 
2.48.1

