diff --git a/dlls/ntdll/unix/fsync.c b/dlls/ntdll/unix/fsync.c
index c3da44e..8d0c76b 100644
--- a/dlls/ntdll/unix/fsync.c
+++ b/dlls/ntdll/unix/fsync.c
@@ -570,11 +570,15 @@ void fsync_init(void)
         ERR("Cannot stat %s\n", config_dir);
 
     if (st.st_ino != (unsigned long)st.st_ino)
-        sprintf( shm_name, "/wine-%lx%08lx-fsync", (unsigned long)((unsigned long long)st.st_ino >> 32), (unsigned long)st.st_ino );
+        sprintf( shm_name, "/data/data/com.termux/files/usr/tmp/wine-%lx%08lx-fsync", (unsigned long)((unsigned long long)st.st_ino >> 32), (unsigned long)st.st_ino );
     else
-        sprintf( shm_name, "/wine-%lx-fsync", (unsigned long)st.st_ino );
+        sprintf( shm_name, "/data/data/com.termux/files/usr/tmp/wine-%lx-fsync", (unsigned long)st.st_ino );
 
-    if ((shm_fd = shm_open( shm_name, O_RDWR, 0644 )) == -1)
+#ifdef __ANDROID__
+    if ((shm_fd = memfd_create(shm_name, 0)) == -1)
+#else
+    if ((shm_fd = memfd_create(shm_name, 0)) == -1)
+#endif
     {
         /* probably the server isn't running with WINEFSYNC, tell the user and bail */
         if (errno == ENOENT)
