diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index 9236974b..fbb4acb4 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -408,8 +408,9 @@ static void kernel_writewatch_softdirty_init(void)
 static void kernel_writewatch_init(void)
 {
     struct uffdio_api uffdio_api;
-
+#if defined __linux__ && !defined __ANDROID__
     uffd_fd = syscall( __NR_userfaultfd, O_CLOEXEC | O_NONBLOCK | UFFD_USER_MODE_ONLY );
+#endif
     if (uffd_fd == -1)
     {
         kernel_writewatch_softdirty_init();