diff --git a/loader/preloader.c b/loader/preloader.c
index f27d6df..c3b6f29 100644
--- a/loader/preloader.c
+++ b/loader/preloader.c
@@ -1513,7 +1513,6 @@ void* wld_start( void **stack )
     /* expose ld.so _r_debug as a separate namespace in r_next */
     ld_so_r_debug = find_symbol( &ld_so_map, "_r_debug", STT_OBJECT );
     if (ld_so_r_debug) _r_debug.r_next = (struct wld_r_debug_extended *)ld_so_r_d>
-    else wld_printf( "_r_debug not found in ld.so\n" );

     _r_debug_state(); /* notify GDB that _r_debug is ready */
diff --git a/server/thread.c b/server/thread.c
index 71c00fb..e85df8f 100644
--- a/server/thread.c
+++ b/server/thread.c
@@ -811,7 +811,7 @@ static void apply_thread_priority( struct thread *thread, int >

     /* FIXME: handle REALTIME class using SCHED_RR if possible, for now map it to>
     if (priority_class == PROCESS_PRIOCLASS_REALTIME) base_priority = 15;
-#ifdef HAVE_SETPRIORITY
+#if defined(HAVE_SETPRIORITY) && !defined(__ANDROID__)
     if (nice_limit < 0)
     {
         niceness = get_unix_niceness( base_priority );
